tests/platform_tests/fwutil/test_fwutil.py

```diff
@@ -0,0 +1,119 @@
+            found = True
+            break
+    assert found
+
+def test_fwutil_install_bad_path(duthost, fw_pkg, random_component):
+    # Test fwutil with bad filepath to firmware
+    out = duthost.command("fwutil install chassis component {} fw BAD.pkg".format(random_component), module_ignore_errors=True)
+    found = False
+    pattern = re.compile(r'.*Error: Invalid value for "<fw_path>"*.')
+    for line in out['stderr_lines']:
+        if pattern.match(line):
+            found = True
+            break
+    assert found
+
+def test_fwutil_update_current(duthost, localhost, pdu_controller, fw_pkg, random_component):
+    assert call_fwutil(duthost,
+            localhost,
+            pdu_controller,
+            fw_pkg, 
+            component=random_component)
+
+def test_fwutil_update_next(duthost, localhost, pdu_controller, fw_pkg, random_component, next_image):
+    assert call_fwutil(duthost, 
+            localhost,
+            pdu_controller,
+            fw_pkg, 
+            component=random_component, 
+            next_image=next_image)
+
+def test_fwutil_update_bad_config(duthost, fw_pkg, random_component):
+    versions = show_firmware(duthost)
+    chassis = versions["chassis"].keys()[0] # Only one chassis
+
+    # Test fwutil update with config file without chassis section
+    
+    with open("platform_components.json", "w") as f:
+        json.dump({}, f, indent=4)
+    upload_platform(duthost, {})
+    found = False
+    out = duthost.command("fwutil update chassis component {} fw -y".format(random_component), module_ignore_errors=True)
+    pattern = re.compile(r'.*Error: Failed to parse "platform_components.json": invalid platform schema*.')
+    for line in out["stdout_lines"]:
+        if pattern.match(line):
+            found = True
+            break
+    assert found
+
+    # Test fwutil update with config file without component section
+    with open("platform_components.json", "w") as f:
+        json.dump({"chassis":{chassis:{}}}, f, indent=4)
+    upload_platform(duthost, {})
+    found = False
+    out = duthost.command("fwutil update chassis component {} fw -y".format(random_component), module_ignore_errors=True)
+    pattern = re.compile(r'.*Error: Failed to parse "platform_components.json": invalid chassis schema*.')
+    for line in out["stdout_lines"]:
+        if pattern.match(line):
+            found = True
+            break
+    assert found
+
+    # Test fwutil update with config file with version of type dict
+    with open("platform_components.json", "w") as f:
+        json.dump({"chassis":{chassis:{"component":{random_component:{"version":{"version":"ver"}}}}}}
+                , f, indent=4)
+    upload_platform(duthost, {})
+    found = False
+    out = duthost.command("fwutil update chassis component {} fw -y".format(random_component), module_ignore_errors=True)
+    pattern = re.compile(r'.*Error: Failed to parse "platform_components.json": invalid component schema*.')
```

Comment:
@alexrallen why do we need to iterate over each line? Isn't it match the multiline case?

Reply:
@alexrallen why do we need this `invalid component schema*.`? e.g, what's the logic behind of this `a*.`?

Reply:
The regular expressions are taken straight from the existing fwutil test cases. They also do function so I am hesitant to change them.

Reply:
Hrm this is just usually how I handle regex, I know some implementations have different ways of going about multi-line matching, splitting it into single lines often takes the guesswork out of it and makes it really clear the scope of the match.